### PR TITLE
654 loopcontrol handle new status hex value from boards

### PR DESCRIPTION
**Pull request by LarsWithCA**

### CODE REVIEW

File: CA_DataUploaderLib/IOconf/IOconfOut230Vac.cs [hunk 1]
```diff
@@ -46,25 +46,25 @@ public SwitchBoardResponseParser (bool expectCommandConfirmations)
...
                 if (numbers.Count < 4)
-                    return null; //invalid line, must have at least 4 currents
+                    return (null, 0); //invalid line, must have at least 4 currents
                 if (numbers.Count == 4)
                     numbers.Add(10000);//some versions did not report temperature
-                return numbers;
+                return (numbers, status);
             }
 
             public override bool MatchesValuesFormat(string line) => _oldswitchBoxCurrentsRegex.IsMatch(line);
```
Comment (LarsWithCA) [inline CA_DataUploaderLib/IOconf/IOconfOut230Vac.cs:70]: @freddyrios should this line be:
`public override bool MatchesValuesFormat(string line) => _oldswitchBoxCurrentsRegex.IsMatch(line) || base.MatchesValuesFormat(line);`
?

File: CA_DataUploaderLib/BaseSensorBox.cs [hunk 2]
```diff
@@ -542,8 +547,8 @@ private async Task<(bool, string?)> TryReadLineWithStallDetection(MCUBoard board
             }
         }
 
-        /// <returns>the list of doubles, otherwise <c>null</c></returns>
-        protected virtual List<double>? TryParseAsDoubleList(MCUBoard board, string line) =>
+        /// <returns>the list of doubles (potentially <c>null</c>) and a status value</returns>
```
Comment (freddyrios) [inline CA_DataUploaderLib/BaseSensorBox.cs:550]: null = failure to parse the line as a double list e.g. we don't have a separate boolean like in the common TryParse pattern

File: CA_DataUploaderLib/IOconf/BoardSettings.cs [hunk 3]
```diff
@@ -35,19 +36,25 @@ public void SetCalibrationAtIndex(string defaultCalibration, char value, int ind
...
-                if (!_hasCommaSeparatedNumbers.IsMatch(line))
-                    return null;
+                var status = 0U;
+                var lineMatch = _hasCommaSeparatedNumbers.Match(line);
+                if (!lineMatch.Success)
+                    return (null, status);
 
-                return line.Split(",".ToCharArray())
+                var statusString = lineMatch.Groups["Status"].Value.Trim(',', ' ');
+                if (!string.IsNullOrEmpty(statusString) && statusString.StartsWith("0x"))
```
Comment (freddyrios) [inline CA_DataUploaderLib/IOconf/BoardSettings.cs:50]: we should use the regex to give you exactly the part of the string you want

something like 

`(?:,\s*0x(?<Status>[0-9a-fA-F]+))?`

File: CA_DataUploaderLib/IOconf/BoardSettings.cs [hunk 4]
```diff
@@ -35,19 +36,25 @@ public void SetCalibrationAtIndex(string defaultCalibration, char value, int ind
...
+                    uint.TryParse(statusString[2..], NumberStyles.HexNumber, CultureInfo.InvariantCulture, out status);
+
+                return (line.Split(",".ToCharArray())
```
Comment (freddyrios) [inline CA_DataUploaderLib/IOconf/BoardSettings.cs:53]: we should use the regex to get the line without the status

File: CA_DataUploaderLib/IOconf/IOconfOut230Vac.cs (hunk 1 above)
Comment (freddyrios) [reply CA_DataUploaderLib/IOconf/IOconfOut230Vac.cs:70]: yes, as that's the value lines we are allowing via TryParseAsDoubleList

**Review by freddyrios — APPROVED** [review]

File: CA_DataUploaderLib/IOconf/IOconfOut230Vac.cs (hunk 1 above)
Comment (LarsWithCA) [reply CA_DataUploaderLib/IOconf/IOconfOut230Vac.cs:70]: @freddyrios  how could it have worked up until now?

File: CA_DataUploaderLib/IOconf/BoardSettings.cs (hunk 4 above)
Comment (LarsWithCA) [reply CA_DataUploaderLib/IOconf/BoardSettings.cs]: done

File: CA_DataUploaderLib/BaseSensorBox.cs (hunk 2 above)
Comment (LarsWithCA) [reply CA_DataUploaderLib/BaseSensorBox.cs]: I updated the text to say this.

File: CA_DataUploaderLib/IOconf/IOconfOut230Vac.cs (hunk 1 above)
Comment (freddyrios) [reply CA_DataUploaderLib/IOconf/IOconfOut230Vac.cs:70]: @LarsWithCA the method is only used in MCUBoard.SkipExtraHeaders where a failure would only result in unnecesarily skipping some value lines.

On initial connection it is not really used for this board due to the setting saying 8 expected header lines (we only skip above 8).

It only leaves reconnects and an extra delay to connect/get values is very likely to be ignored.

Comment (LarsWithCA) [reply CA_DataUploaderLib/IOconf/IOconfOut230Vac.cs]: fixed

**Pull request merged**